sample_size/multiple_testing.py

```diff
@@ -37,23 +48,23 @@ def get_multiple_sample_size(self, lower: float, upper: float, depth: int = 0) -
 
         Returns minimum required sample size per cohort
         """
-        max_recursion_depth: int = 20
-        epsilon: float = 0.025  # TODO(any): make this configurable by users ML-5429
 
         if depth > max_recursion_depth:
             raise RecursionError(f"Couldn't find a sample size that satisfies the power you requested: {self.power}")
 
-        candidate = int(lower + (upper - lower) / 2)
-        expected_power = self._expected_average_power(candidate)
-        if np.isclose(self.power, expected_power, epsilon):
+        candidate = int(np.sqrt(lower * upper))
```

Comment:
Why is this better than the average of lower and upper?

Reply:
from the original notebook:

> One thing to note is that because we use Bonferroni correction to calculate an an initial upper bound, our upper estimates are are usually very large (sometimes in the tens of billions). We can improve the rate at which we converge on the true sample-size by switching from using the arithmetic mean to calculate our candidate sample-size to using our a geometric mean (so our candidate sample sizes are closer to the lower bound).